lib/winston/transports/file.js

```diff
@@ -685,4 +687,12 @@ module.exports = class File extends TransportStream {
       );
     });
   }
+
+  _createLogDirIfNotExist(dirPath) {
+    /* eslint-disable no-sync */
+    if (!fs.existsSync(dirPath)) {
```

Comment:
In general we should avoid sync functions but in this case I would agree with you that we should just do this synchronously, it's such a cheap operation...

Reply:
Yes, it will be invoked once for each launch. It is cheap.